### DISCUSSION TITLE
Anybody else encountering a Big number issue in Delegating?

### DISCUSSION THREAD

**Discussion by benjnp** [ping-pub/explorer#487] [Blockchains]
Hi,

We're trying to run Ping for our testnet project but in the current version it uses the base denom in delegating which means that in order to delegate let's say 100K tokens, we'll have to enter so many zeroes causing an error like this one below "cannot unmarshal big int..."
<img width="494" alt="image" src="https://github.com/ping-pub/explorer/assets/32241282/7951fb0b-5455-4efe-ad40-7fa552a505ef">

Anybody got a solution on this one?

Thanks

**Comment by liangping** [comment]
That's the issue on backend

**Reply by benjnp** [reply]
Backend on the chain config or with Ping's backend?

**Reply by liangping** [reply]
I would suggest that you add denom meta which will let you to input a smaller number for amount.

**Comment by liangping** [comment]
Sorry, it's trying to convert '1e24' to int.

**Reply by benjnp** [reply]
I actually want to be allowed to input bigger values. I noticed with other chains in Ping, they're allowed to use the base and 10^18 denom so for Axelar it allows you to choose between uaxl and axl. In our case, our token is BCX but our delegation dialog only allows us to use abcx and not bcx.

**Comment by liangping** [comment]
Yeah. we currently load the denom metadata from chain repository, if metadata is loaded, the converter will work . will read it from chains later(many chains do not have correct denom metadata).

**Reply by benjnp** [reply]
Ok so this is more of code in the chain not the json file that we put in the chains/mainnet folder?

**Reply by benjnp** [reply]
May I know a sample of denom metadata from an existing chain repository? Thanks

**Comment by liangping** [comment]
https://github.com/cosmos/chain-registry/blob/master/cosmoshub/assetlist.json

I am still seeking a better solution for this.

**Reply by benjnp** [reply]
Got it. By the way, we'd like to add our project to Ping.pub but because of this denom issue, we can only stake below 1000 because there's 18 zeroes after it and I saw that we need to test the delegation to make sure that it is working ok. So in our case, it is working except for the denom only showing aBCX while our main token is BCX. If we'll be able to select BCX then we don't need to enter the 18 zeroes and we can enter big numbers (more than 1000). Do we have to wait for the solution before we can submit our project?


Thanks

**Comment by liangping** [comment]
No. You can submit as you want. I haven't figured out the best solution for now. But i am thinking changing the number to bignumber first.&nbsp;



---Original---
From: ***@***.***&gt;
Date: Thu, Aug 17, 2023 22:05 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ping-pub/explorer] Anybody else encountering a Big number issuein Delegating? (Discussion #487)




 
Got it. By the way, we'd like to add our project to Ping.pub but because of this denom issue, we can only stake below 1000 because there's 18 zeroes after it and I saw that we need to test the delegation to make sure that it is working ok. So in our case, it is working except for the denom only showing aBCX while our main token is BCX. If we'll be able to select BCX then we don't need to enter the 18 zeroes and we can enter big numbers (more than 1000). Do we have to wait for the solution before we can submit our project?
 
Thanks

**Reply by benjnp** [reply]
Thanks, we're using an older version of ping for now and it is working just fine.

**Comment by liangping** [comment]
Hey. Could be please add metadata in genesis.&nbsp;
https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-024-coin-metadata.md



Let me know if there is still the question

---Original---
From: ***@***.***&gt;
Date: Sat, Aug 19, 2023 00:16 AM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ping-pub/explorer] Anybody else encountering a Big number issuein Delegating? (Discussion #487)




 
Thanks, we're using an older version of ping for now and it is working just fine.

**Comment by liangping** [comment]
Hi, @benjnp 

Here's final solution for this issue: 

There are two ways to load denom metadata for unit converter.

1. add denoms of your blockchain in genesis file, here's more infomation: 
https://github.com/cosmos/cosmos-sdk/blob/main/docs/architecture/adr-024-coin-metadata.md
2. register denoms on registry: https://github.com/cosmos/chain-registry

Once you have the correct unit converter, you can enter up to a numeric quantity (10^23), which should be enough for most use cases.

**Reply by benjnp** [reply]
Thanks on this. we'll give this one a try.

**Comment by benjnp** [comment]
Hi,

We got the denom working and it can now recognize our denoms properly but
I'm getting another error which is regarding account signing. I'm getting
this in delegation and sending or basically when it is time to sign. It
says that our chain was not found in the registry. Is there another way to
get around it?

[image: image.png]

Thanks and Regards,
Benjamin

**Comment by benjnp** [comment]
Hi,

My apologies, it is working despite the message in the console. I just had
the keplr wallet selected a wrong account.


Thanks

**Comment by benjnp** [comment]
Hi,

I'm doing the tests for us to be able to add the chain to your testnet but
I got an error like this one below. Is this normal? I can't redelegate to
the same validator?
[image: image.png]

Also, in the test there's an item regarding withdrawing validator's
commission where is it located? Is it the same as withdraw rewards?
And by the way, is it normal that it wont work yet to Metamask staking
because our chain is not yet added to your system?


Thanks and Regards,
Benjamin

**Comment by liangping** [comment]
> I'm doing the tests for us to be able to add the chain to your testnet but I got an error like this one below. Is this normal? I can't redelegate to the same validator?

your image does not show up. could reload it.

> Also, in the test there's an item regarding withdrawing validator's
commission where is it located? Is it the same as withdraw rewards?

It's on `commission & reward` panel in validator page.

> And by the way, is it normal that it wont work yet to Metamask staking
because our chain is not yet added to your system?

You use `Wallet Helper` on bottom of left-side bar. to suggest you chain to Metamask.

**Comment by benjnp** [comment]
Hi,

I added the chain manually in Metamask then connect it to Ping and it
recognizes the balance when I go to Dashboard but it seems to be having
problems with signature.
My first error is this one below when I try to send tokens, it actually
prompted Metamask for the signature.
[image: image.png]

Then when I click on Sign, the error becomes like this one below
[image: image.png]

Also I noticed that the chain was not added in Metamask even when I use the
wallet helper, I have to manually add the chain to Metamask but it is fine
just not sure if it will help in troubleshooting.


Thanks and Regards,
Benjamin

**Comment by liangping** [comment]
You screenshot not uploaded. I don't know what's the problem?

**Comment by benjnp** [comment]
Maybe because I attached the image via email so this time I'll upload the image in github.

I added the chain manually in Metamask then connect it to Ping and it recognizes the balance when I go to Dashboard but it seems to be having problems with signature.
My first error is this one below when I try to send tokens, it actually prompted Metamask for the signature.
<img width="1245" alt="image" src="https://github.com/ping-pub/explorer/assets/32241282/bf57026d-c1a7-44d1-9dc0-ad85700121bd">

Then when I click on Sign, the error becomes like this one below
<img width="1431" alt="image" src="https://github.com/ping-pub/explorer/assets/32241282/3cc19d3d-f359-4e8d-ac53-252a6377e233">

Also I noticed that the chain was not added in Metamask even when I use the wallet helper, I have to manually add the chain to Metamask but it is fine just not sure if it will help in troubleshooting.

By the way, since the error is in Metamask wallet and everything is working when Keplr is used, does it mean it can be submitted for PR?


Thanks

**Comment by liangping** [comment]
This is because you have issues on gas simulation. Please check the 'simulate' API.&nbsp;
And Snap only used to Cosmos not EVMOS based block chain



---Original---
From: ***@***.***&gt;
Date: Sat, Sep 16, 2023 22:24 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [ping-pub/explorer] Anybody else encountering a Big number issuein Delegating? (Discussion #487)




 
Maybe because I attached the image via email so this time I'll upload the image in github.
 
I added the chain manually in Metamask then connect it to Ping and it recognizes the balance when I go to Dashboard but it seems to be having problems with signature.
 My first error is this one below when I try to send tokens, it actually prompted Metamask for the signature.
 
 
Then when I click on Sign, the error becomes like this one below
 
 
Also I noticed that the chain was not added in Metamask even when I use the wallet helper, I have to manually add the chain to Metamask but it is fine just not sure if it will help in troubleshooting.
 
By the way, since the error is in Metamask wallet and everything is working when Keplr is used, does it mean it can be submitted for PR?
 
Thanks

**Reply by benjnp** [reply]
But it is working on normal Metamask transfer and dapp usage. Do you know how to check this simulate API? Sorry what do you mean by "Snap only used to Cosmos not EVMOS based block chain" and what is Snap?

**Reply by benjnp** [reply]
Hi,

Is this what you meant as Simulate API? 
https://atlantis-api.blockxnet.com/swagger/#/Service/Simulate

Is there's something wrong with it?


Thanks

**Comment by liangping** [comment]
the first screen shot is simulate error.
the second is signing issue

**Reply by benjnp** [reply]
Although this only happens when Metamask is being used, will this be a blocker in adding our chain?

**Comment by liangping** [comment]
No. You can submit you chain if you want

**Comment by benjnp** [comment]
Thanks, by the way is ping.pub having some updates? It has been inaccessible for me for the past few days. I also tried different connection and devices but to no avail.

**Reply by liangping** [reply]
It's available on my side. which city are you in?

**Comment by benjnp** [comment]
Oh right, I tried using VPN and it worked. I'm from the Philippines.

**Reply by benjnp** [reply]
Hi,

It seems the Testnet folder is missing
<img width="1045" alt="image" src="https://github.com/ping-pub/explorer/assets/32241282/34b27853-1ae3-459a-b92b-e4f476d934ef">

Thanks

**Reply by liangping** [reply]
it's moved to https://github.com/ping-pub/ping.pub

**Reply by benjnp** [reply]
Thanks got it. But I'm not allowed to push a new branch for the PR.

**Reply by benjnp** [reply]
Hi,

To submit a json file, we're going to create a branch for PR right? I tried to push the json but it says unauthorized.

Thanks

**Comment by benjnp** [comment]
Hi,

Just following up on this.

Thanks

**Reply by benjnp** [reply]
Hi,

Good Day. Just checking to see if you can help us on this matter. 


Thanks

**Comment by liangping** [comment]
Oh，man, you have to use Pull Request.

**Reply by benjnp** [reply]
Oh apologies, I thought we were using the forked version. Thanks. Done submitting.

**Comment by benjnp** [comment]
Hi,

For the signing issue when delegating using Metamask, the error is this
<img width="486" alt="image" src="https://github.com/ping-pub/explorer/assets/32241282/bb9763f3-676a-4308-9695-b3a1a7530c14">


Do you know the solution that we can do on our end? Will the Cosmos Registry registration solve this?


Thanks
